### PR TITLE
Task 15. Create Dashboard

### PR DESCRIPTION
**Pull request by Julia1910**

### CODE REVIEW

**Review by Barburon — APPROVED** [review]

File: src/main/java/com/cursor/view/Actions.java [hunk 1]
```diff
@@ -113,6 +117,72 @@ public void showActionsMenu() {
...
+            System.out.println("Wrong input, please use numbers");
+        }
+    }
+
+    private User getUser() {
+        System.out.println("Enter, please, User's ID");
+        int UserID = scanner.nextInt();
+        return userService.findById(UserID);
+    }
+
```
Comment (ShvedovIllia) [inline src/main/java/com/cursor/view/Actions.java:169]: remove empty line

File: src/main/java/com/cursor/view/Actions.java [hunk 2]
```diff
@@ -113,6 +117,72 @@ public void showActionsMenu() {
...
+        System.out.println("'2' - Show spent time");
+        System.out.println("'3' - Find most time-expensive ticket");
+        System.out.println("'4' - Show tickets for special user");
+        System.out.println("'5' - Show System Statistics");
+        System.out.println("'6' - Show User Statistics");
+        System.out.println("'7' - Find tickets with special status");
+        System.out.println("'8' - Find tickets with special priority");
+        System.out.println("'0' - Back to Ticket's menu");
+    }
+
```
Comment (ShvedovIllia) [inline src/main/java/com/cursor/view/Actions.java:184]: remove empty line

File: src/main/java/com/cursor/service/DashboardImpl.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        return tickets
+                .stream()
+                .filter(ticket -> ticket.getStatus().equals(status))
+                .collect(Collectors.teeing(
+                        Collectors.counting(),
+                        Collectors.toList(),
+                        (number, sortTickets) -> status.toString() + " " + number +
+                                " tickets:  " + sortTickets.toString() + "\n"));
+    }
+
```
Comment (ShvedovIllia) [inline src/main/java/com/cursor/service/DashboardImpl.java:118]: remove empty lines here and below

**Review by ShvedovIllia — CHANGES_REQUESTED** [review]

**Review by AmorBielyi — APPROVED** [review]

**Review by ShvedovIllia — APPROVED** [review]

File: src/main/java/com/cursor/service/DashboardImpl.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+public class DashboardImpl implements Dashboard {
+    private TicketDao tickets = TicketDao.getInstance();
+    private UserDao users = UserDao.getInstance();
+
+    @Override
+    public int getTotalTime(User user) {
+        int time = 0;
+        for (Ticket ticket : tickets.getAll()) {
+            if (ticket.getAssignee().equals(user)) {
```
Comment (lazermann111) [inline src/main/java/com/cursor/service/DashboardImpl.java:25]: Its a common situation in real Jira when ticket doesnt have an assignee (unassigned ticket), so maybe it makes sesnse to add null check here

File: src/main/java/com/cursor/service/DashboardImpl.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+                time += ticket.getTimeEstimated();
+            }
+        }
+        return time;
+    }
+
+    @Override
+    public int getSpentTime(User user) {
+        int time = 0;
+        for (Ticket ticket : tickets.getAll()) {
```
Comment (lazermann111) [inline src/main/java/com/cursor/service/DashboardImpl.java:35]: Its a common situation in real Jira when ticket doesnt have an assignee (unassigned ticket), so maybe it makes sesnse to add null check here

**Review by lazermann111 — APPROVED** [review]
Approved with minor comment

**Pull request merged**